src/kernels/composable_kernel/src/kernel_wrapper/gridwise_convolution_backward_data_implicit_gemm_v1r1_xdlops_nchw_kcyx_nkhw.cpp

```diff
@@ -53,156 +66,153 @@ extern "C" __global__
-#else
-    static_assert(false, "wrong! Only fp32, fp16 and bfp16 are supported.");
-#endif
-
-    // C matrix
-    constexpr auto GemmMPerWave = CK_PARAM_GEMM_M_PER_WAVE;
-    constexpr auto GemmNPerWave = CK_PARAM_GEMM_N_PER_WAVE;
-
-    constexpr index_t GemmThreadGemmDataPerReadM = 1;
-    constexpr index_t GemmThreadGemmDataPerReadN = 1;
-
-#if MIOPEN_USE_FP32
-    constexpr auto gridwise_conv_bwd_data =
-        GridwiseConvolutionBackwardDataImplicitGemm_v1r1_xdlops_nchw_kcyx_nkhw<
-            GridSize,
-            BlockSize,
-            FLOAT,
-            FLOAT_ACCUM,
-            decltype(in_nchw_desc),
-            decltype(wei_kcyx_desc),
-            decltype(out_nkhw_desc),
-            ConvStrides,
-            ConvDilations,
-            InLeftPads,
-            InRightPads,
-            GemmMPerBlock,
-            GemmNPerBlock,
-            GemmKPerBlock,
-            GemmMPerWave,
-            GemmNPerWave,
-            GemmThreadGemmDataPerReadM,
-            GemmThreadGemmDataPerReadN,
-            GemmABlockCopyThreadSliceLengths_GemmK_GemmM,
-            GemmABlockCopyThreadClusterLengths_GemmK_GemmM,
-            GemmABlockCopySrcDataPerRead_GemmM,
-            GemmABlockCopyDstDataPerWrite_GemmM,
-            GemmBBlockCopyThreadSliceLengths_GemmK_GemmN,
-            GemmBBlockCopyThreadClusterLengths_GemmK_GemmN,
-            GemmBBlockCopySrcDataPerRead_GemmN,
-            GemmBBlockCopyDstDataPerWrite_GemmN>{};
-
-    gridwise_conv_bwd_data.Run(p_in_global, p_wei_global, p_out_global);
-
-#elif MIOPEN_USE_FP16 || MIOPEN_USE_BFP16
-    constexpr auto gridwise_conv_bwd_data =
-        GridwiseConvolutionBackwardDataImplicitGemm_v1r1_xdlops_f16_bfp16_nchw_kcyx_nkhw<
-            GridSize,
-            BlockSize,
-            FLOAT,
-            FLOAT_ACCUM,
-            decltype(in_nchw_desc),
-            decltype(wei_kcyx_desc),
-            decltype(out_nkhw_desc),
-            ConvStrides,
-            ConvDilations,
-            InLeftPads,
-            InRightPads,
-            GemmMPerBlock,
-            GemmNPerBlock,
-            GemmKPerBlock,
-            GemmKPACK,
-            GemmMPerWave,
-            GemmNPerWave,
-            GemmThreadGemmDataPerReadM,
-            GemmThreadGemmDataPerReadN,
-            GemmABlockCopyThreadSliceLengths_GemmK_GemmM_GemmKPACK,
-            GemmABlockCopyThreadClusterLengths_GemmK_GemmM_GemmKPACK,
-            GemmABlockCopySrcDataPerRead_GemmM,
-            GemmABlockCopyDstDataPerWrite_GemmKPACK,
-            GemmBBlockCopyThreadSliceLengths_GemmK_GemmN_GemmKPACK,
-            GemmBBlockCopyThreadClusterLengths_GemmK_GemmN_GemmKPACK,
-            GemmBBlockCopySrcDataPerRead_GemmN,
-            GemmBBlockCopyDstDataPerWrite_GemmKPACK>{};
-
-    gridwise_conv_bwd_data.Run(
-        reinterpret_cast<FLOAT_ACCUM*>(p_in_global), p_wei_global, p_out_global);
+    constexpr auto wkgrp_schd_order =
+#if MIOPEN_USE_FP16 || MIOPEN_USE_BFP16
```

Comment:
fp16 and bfp16 has different behavior. Is this intended?

Reply:
Yes, fp32 vs fp16/bfp16. I didn't see any perf gain for wkgrp scheduling with fp32.

Reply:
Memory access related (buffer walks) code should be identical for FP16/BF16, I think.